### PR TITLE
[Android][1.2] Fix placeholders for Input.Time and Input.Date

### LINKED ISSUE #4484: [Android] Placeholder values of input components are not applied
# Platform

What platform is your issue or question related to? (Delete other platforms).

* Android

# Author or host

Microsoft Teams

# Version of SDK

1.2.10

# Details

As per the schema, Input components do have `placeholder` property which are to be shown when no selection has been made.

Placeholder properties aren't utilised on Android for Input.Number, Input.Date, Input.Time. 

Here is the experience:

| Android Experience | Desktop OOB Experience | Desktop Teams Experience |  iOS Teams Experience
|-----------|:-----------:|-----------:|  -----------:|  
| Screenshot 2020-07-24 at 4 05 43 PM | Screenshot 2020-07-24 at 4 06 29 PM | Screenshot 2020-07-24 at 4 06 52 PM | image |

Here is the card JSON:

```
{
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "type": "AdaptiveCard",
    "version": "1.2",
    "body": [
        {
            "type": "TextBlock",
            "text": "Date Input"
        },
        {
            "type": "Input.Date",
            "id": "date",
            "placeholder": "Enter a date"
        },
        {
            "type": "Input.Number",
            "id": "number",
            "placeholder": "Enter a number within 1-10",
            "min": 1,
            "max": 10
        },
        {
      "type": "Input.Text",
      "id": "input1",
      "placeholder": "enter comment",
      "maxLength": 500
    },
    {
      "type": "Input.Time",
      "id": "time",
      "min": "09:00",
      "max": "17:00",
      "placeholder": "Enter time between 9-17"
    }
    ]
}
```

cc: @VikrantSingh01, @anbah34, @shbhadra

### PR DESCRIPTION
**Pull request by almedina-ms**
## Related Issue
Fixes #4484 

## Description
Adds validations to check if the value is empty or not valid to place a placeholder as other platforms do

## How Verified
The fix was verified manually

image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4794)

### CODE REVIEW

**Comment by shalinijoshi19** [comment]
@almedina-ms am assuming we are fixing this also for our main line branch and not just the 1.2 servicing branch yes?

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RendererUtil.java [hunk 1]
```diff
@@ -33,6 +33,12 @@ public static Calendar getDate(String s)
         return calendar;
     }
 
+    public static boolean isValidTime(String s)
```
Comment (shalinijoshi19) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RendererUtil.java:36]: Question: How come no equivalent check for Date values? Do we validate that earlier on or elsewhere ? Also what date formats are we allowing again - per our schema looks like - "YYYY-MM-DD" ?

**Review by shalinijoshi19 — APPROVED** [review]

**Comment by almedina-ms** [comment]
> @almedina-ms am assuming we are fixing this also for our main line branch and not just the 1.2 servicing branch yes?

This has been fixed in 2.x already, we are "backporting" the fix for Teams

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RendererUtil.java (hunk 1 above)
Comment (almedina-ms) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RendererUtil.java:41]: added the mirrored change

**Comment by almedina-ms** [comment]
This is no longer required

**Pull request closed without merging**